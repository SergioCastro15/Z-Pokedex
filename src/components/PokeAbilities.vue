<template>
  <div>
    <!-- Single button triggers two "<b-collapse>" components -->
    <b-button class="mb-3" @click="showPowers">Show powers</b-button>

    <b-list-group v-if="showAbility">
      <b-list-group-item
        v-for="(abilitie, index) in abilities"
        :key="index"
      >
        {{abilitie.ability.name}}
      </b-list-group-item>
    </b-list-group>
  </div>
</template>

<script>
export default {
  name: 'PokeAbilities',
  data() {
    return {
      showAbility: false,
    };
  },
  props: {
    abilities: {
      type: Array,
      required: false,
    },
  },
  methods: {
    showPowers() {
      this.showAbility = !this.showAbility;
    },
  },
};
</script>
