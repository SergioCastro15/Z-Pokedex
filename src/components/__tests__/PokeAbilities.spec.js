import { shallowMount } from '@vue/test-utils';
import PokeAbilities from '../PokeAbilities.vue';

describe('PokeAbilities component', () => {
  it('render', () => {
    const wrapper = shallowMount(PokeAbilities, {
      propsData: {
        abilities: [
          {
            ability: {
              name: 'Pikachu',
            },
          },
          {
            ability: {
              name: 'Charmander',
            },
          },
        ],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should show the powers of current pokemon', () => {
    const localThis = {
      showAbility: false,
    };
    PokeAbilities.methods.showPowers.call(localThis);
    expect(localThis.showAbility).toBe(true);
  });
});
