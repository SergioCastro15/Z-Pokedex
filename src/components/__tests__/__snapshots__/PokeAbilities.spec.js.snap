// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeAbilities component render 1`] = `
<div>
  <b-button
    class="mb-3"
  >
    Show powers
  </b-button>
   
  <!---->
</div>
`;
